### ISSUE TITLE
jBallerina sprint - Week 14 - 2019

### ISSUE THREAD

**Issue by Kishanthan** [ballerina-platform/ballerina-lang#14752]
- [x] Object code generation - @SupunS
    - Basically object end to end scenarios (new, attached function Invocation, etc)
- [x] Concurrency and futures- Identify the high level tasks and decide on what tasks to complete for each week - @vinok88 
    - Design review on futures approach
- [x] BIR Positions Code generation - @warunalakshitha  (2 Days)
- [x] Built-in methods (freeze, clone) - @warunalakshitha 
- [ ] Recursive types to work with BIR gen - @manuranga 
    - Records
    - User defined Error
    - Objects
- [x] Design approach on endpoints, services, resources - @Kishanthan 
    - Basically the new connector API
